### PR TITLE
Add Percival

### PR DESCRIPTION
**Pull request by ocots**

### CODE REVIEW

File: docs/src/tutorial-nlp.md [hunk 1]
```diff
@@ -48,22 +59,28 @@ You can extract this raw NLP problem  with the [`get_nlp`](@ref) method.
 nlp = get_nlp(docp)
```
Comment (jbcaillau) [inline docs/src/tutorial-nlp.md:59]: @ocots this syntax has been changed in CTDirect 0.10 (and will be in OptimalControl next release) ; @PierreMartinon do you confirm it should be `docp, nlp = direct_transcription(...)`?

**Review by jbcaillau — COMMENTED** [review]
@ocots nice to have added Percival in the doc 👍🏽

**Comment by ocots** [comment]
> @ocots nice to have added Percival in the doc 👍🏽

Yes but it does not seem to work well. I have used the solution from ipopt as initial guess but no improvement compared to default initial guess.

File: docs/src/tutorial-nlp.md (hunk 1 above)
Comment (ocots) [reply docs/src/tutorial-nlp.md:59]: @PierreMartinon has already changed this file. We will have to merge both updates.

**Comment by PierreMartinon** [comment]
@ocots maybe we merge this one first then update the 0.10.0 one ?

**Comment by ocots** [comment]
As you want. I can also update it later.

**Comment by PierreMartinon** [comment]
> As you want. I can also update it later.

Done. All yours :D

**Comment by jbcaillau** [comment]
@ocots are you waiting to solve the conflicts above to release `0.10.0`?

**Comment by ocots** [comment]
No I am just waiting to dry 🏖

**Comment by jbcaillau** [comment]
> @jbcaillau No I am just waiting to dry 🏖

https://github.com/control-toolbox/OptimalControl.jl/issues/251#issuecomment-2262377015

**Comment by ocots** [comment]
Ok I do it asap.

**Comment by PierreMartinon** [comment]
I fixed the tutorial conflict (2 calls to get_nlp removed)

**Comment by ocots** [comment]
> I fixed the tutorial conflict (2 calls to get_nlp removed) By the way, I can look into having percival as a solver option for `CTDirect`. This would be a package extension as for ipopt. Ideally we would tailor the error message for missing packages (eg 'Please execute use NLPModelsIpopt / Percival(?)' when nlp_solver = ipopt / percival etc), cf the error types discussion.

Could be nice. Have you seen that it does not perform well on the example?

**Comment by PierreMartinon** [comment]
Not yet, I'll try the small benchmark series

**Pull request merged**